### ISSUE TITLE
Allow options to be supplied to IocConstruct

### ISSUE THREAD

**Issue by softlion** [MvvmCross/MvvmCross#534]
Hi,
i would like a better IocConstruct method, so it can be used to build third parties object, like the one described in the answer here : http://stackoverflow.com/questions/20823427/mvvmcross-sqlite-community-plugin-wont-add-to-xamarin-studio-project/20825736?noredirect=1#comment31243538_20825736

Ideally, i would like to be able to call a constructor with parameters using an anonymous object to fill other parameter values, the same way ASP.NET MVC does with its actions:

```
var sqlConnection = Mvx.IocConstruct<SQLiteConnection>({path="mydbpath", mode=Mode.ReadOnly});
```

Or better, have the parameters set at registration time:

```
Mvx.RegisterType<ISQLitePlatform, SQLitePlatformIOS>().WithParameters({path="mydbpath", mode=Mode.ReadOnly});
```

**Comment by MKuckert** [comment]
I have an implementation for the first case to pass arguments to `IocConstruct`: `var sqlConnection = Mvx.IocConstruct<SQLiteConnection>({path="mydbpath", mode=Mode.ReadOnly});`

https://github.com/SALT-AND-PEPPER/MvvmCross/commit/509b15c0e1e24e52310d281d1873e30181782e32

It's currently based on the old 5.7.0 release because I can't successfully open the current develop branch in VS4Mac (complains when resolving packages about projects not being compliant to .Net Standard 🤪). I will re-base on 6.0.0 if somebody can guide me here.

**Comment by nmilcoff** [comment]
This was fixed in #2814
